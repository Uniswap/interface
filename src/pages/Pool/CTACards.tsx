import { Trans } from '@lingui/macro'
import { useWeb3React } from '@web3-react/core'
import { AutoColumn } from 'components/Column'
import { getChainInfoOrDefault } from 'constants/chainInfo'
import styled from 'styled-components/macro'
import { ThemedText } from 'theme'

import { ExternalLink } from '../../theme'

const CTASection = styled.section`
  display: grid;
  grid-template-columns: 1fr 1fr;
  gap: 8px;
  opacity: 0.8;

  ${({ theme }) => theme.deprecated_mediaWidth.deprecated_upToSmall`
    grid-template-columns: auto;
    grid-template-rows: auto;
  `};
`

const CTA1 = styled(ExternalLink)`
  padding: 16px;
  border-radius: 20px;
  display: flex;
  flex-direction: column;
  position: relative;
  justify-content: center;
  align-items: center;
  overflow: hidden;
  border: 1px solid ${({ theme }) => theme.deprecated_bg3};

  * {
    color: ${({ theme }) => theme.textPrimary};
    text-decoration: none !important;
  }

  :hover {
    border: 1px solid ${({ theme }) => theme.deprecated_bg4};

    text-decoration: none;
    * {
      text-decoration: none !important;
    }
  }
`

const CTA2 = styled(ExternalLink)`
  position: relative;
  overflow: hidden;
  padding: 16px;
  border-radius: 20px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  border: 1px solid ${({ theme }) => theme.deprecated_bg3};

  * {
    color: ${({ theme }) => theme.textPrimary};
    text-decoration: none !important;
  }

  :hover {
    border: 1px solid ${({ theme }) => theme.deprecated_bg4};
    text-decoration: none !important;
    * {
      text-decoration: none !important;
    }
  }
`

const HeaderText = styled(ThemedText.DeprecatedLabel)`
  align-items: center;
  display: flex;

  font-weight: 400;
  font-size: 16px;
  ${({ theme }) => theme.deprecated_mediaWidth.deprecated_upToMedium`
    font-size: 16px;
  `};
`

const ResponsiveColumn = styled(AutoColumn)`
  grid-template-columns: 1fr;
  width: 100%;
  gap: 8px;

  ${({ theme }) => theme.deprecated_mediaWidth.deprecated_upToMedium`
    gap: 8px;
  `};
  justify-content: space-between;
`

export default function CTACards() {
  const { chainId } = useWeb3React()
  const { infoLink } = getChainInfoOrDefault(chainId)

  return (
    <CTASection>
      <CTA1 href="https://support.uniswap.org/hc/en-us/categories/8122334631437-Providing-Liquidity-">
        <ResponsiveColumn>
          <HeaderText>
            <Trans>Learn about providing liquidity</Trans> ↗
          </HeaderText>
          <ThemedText.DeprecatedBody fontWeight={400} style={{ alignItems: 'center', display: 'flex' }}>
            <Trans>Check out our v3 LP walkthrough and migration guides.</Trans>
          </ThemedText.DeprecatedBody>
        </ResponsiveColumn>
      </CTA1>
      <CTA2 data-testid="cta-infolink" href={infoLink + 'pools'}>
        <ResponsiveColumn>
          <HeaderText style={{ alignSelf: 'flex-start' }}>
            <Trans>Top pools</Trans> ↗
          </HeaderText>
          <ThemedText.DeprecatedBody fontWeight={400} style={{ alignSelf: 'flex-start' }}>
            <Trans>Explore Uniswap Analytics.</Trans>
          </ThemedText.DeprecatedBody>
        </ResponsiveColumn>
      </CTA2>
    </CTASection>
  )
}
