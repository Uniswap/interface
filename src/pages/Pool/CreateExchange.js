import React, { useState, useEffect } from 'react'
import { withRouter } from 'react-router'
import { createBrowserHistory } from 'history'
import { ethers } from 'ethers'
import styled from 'styled-components'
import { useTranslation } from 'react-i18next'
import ReactGA from 'react-ga'

import { useWeb3React, useFactoryContract } from '../../hooks'
import { Button } from '../../theme'
import AddressInputPanel from '../../components/AddressInputPanel'
import OversizedPanel from '../../components/OversizedPanel'
import { useTokenDetails } from '../../contexts/Tokens'
import { useTransactionAdder } from '../../contexts/Transactions'

const SummaryPanel = styled.div`
  ${({ theme }) => theme.flexColumnNoWrap};
  padding: 1rem 0;
`

const ExchangeRateWrapper = styled.div`
  ${({ theme }) => theme.flexRowNoWrap};
  align-items: center;
  color: ${({ theme }) => theme.doveGray};
  font-size: 0.75rem;
  padding: 0.25rem 1rem 0;
`

const ExchangeRate = styled.span`
  flex: 1 1 auto;
  width: 0;
  color: ${({ theme }) => theme.doveGray};
`

const CreateExchangeWrapper = styled.div`
  color: ${({ theme }) => theme.doveGray};
  text-align: center;
  margin-top: 1rem;
  padding-top: 1rem;
`

const SummaryText = styled.div`
  font-size: 0.75rem;
  color: ${({ error, theme }) => error && theme.salmonRed};
`

const Flex = styled.div`
  display: flex;
  justify-content: center;
  padding: 2rem;

  button {
    max-width: 20rem;
  }
`

function CreateExchange({ location, params }) {
  const { t } = useTranslation()
  const { account } = useWeb3React()

  const factory = useFactoryContract()

  const [tokenAddress, setTokenAddress] = useState({
    address: params.tokenAddress ? params.tokenAddress : '',
    name: ''
  })
  const [tokenAddressError, setTokenAddressError] = useState()

  const { name, symbol, decimals, exchangeAddress } = useTokenDetails(tokenAddress.address)
  const addTransaction = useTransactionAdder()

  // clear url of query
  useEffect(() => {
    const history = createBrowserHistory()
    history.push(window.location.pathname + '')
  }, [])

  // validate everything
  const [errorMessage, setErrorMessage] = useState(!account && t('noWallet'))
  useEffect(() => {
    if (tokenAddressError) {
      setErrorMessage(t('invalidTokenAddress'))
    } else if (symbol === undefined || decimals === undefined || exchangeAddress === undefined) {
      setErrorMessage()
    } else if (symbol === null) {
      setErrorMessage(t('invalidSymbol'))
    } else if (decimals === null) {
      setErrorMessage(t('invalidDecimals'))
    } else if (exchangeAddress !== ethers.constants.AddressZero) {
      setErrorMessage(t('exchangeExists'))
    } else if (!account) {
      setErrorMessage(t('noWallet'))
    } else {
      setErrorMessage(null)
    }

    return () => {
      setErrorMessage()
    }
  }, [tokenAddress.address, symbol, decimals, exchangeAddress, account, t, tokenAddressError])

  async function createExchange() {
    const estimatedGasLimit = await factory.estimate.createExchange(tokenAddress.address)

    factory.createExchange(tokenAddress.address, { gasLimit: estimatedGasLimit }).then(response => {
      ReactGA.event({
        category: 'Transaction',
        action: 'Create Exchange'
      })

      addTransaction(response)
    })
  }

  const isValid = errorMessage === null

  return (
    <>
      <AddressInputPanel
        title={t('tokenAddress')}
        initialInput={
          params.tokenAddress
            ? { address: params.tokenAddress }
            : { address: (location.state && location.state.tokenAddress) || '' }
        }
        onChange={setTokenAddress}
        onError={setTokenAddressError}
      />
      <OversizedPanel hideBottom>
        <SummaryPanel>
          <ExchangeRateWrapper>
            <ExchangeRate>{t('name')}</ExchangeRate>
            <span>{name ? name : ' - '}</span>
          </ExchangeRateWrapper>
          <ExchangeRateWrapper>
            <ExchangeRate>{t('symbol')}</ExchangeRate>
            <span>{symbol ? symbol : ' - '}</span>
          </ExchangeRateWrapper>
          <ExchangeRateWrapper>
            <ExchangeRate>{t('decimals')}</ExchangeRate>
            <span>{decimals || decimals === 0 ? decimals : ' - '}</span>
          </ExchangeRateWrapper>
        </SummaryPanel>
      </OversizedPanel>
      <CreateExchangeWrapper>
        <SummaryText>{errorMessage ? errorMessage : t('enterTokenCont')}</SummaryText>
      </CreateExchangeWrapper>
      <Flex>
        <Button disabled={!isValid} onClick={createExchange}>
          {t('createExchange')}
        </Button>
      </Flex>
    </>
  )
}

export default withRouter(CreateExchange)
