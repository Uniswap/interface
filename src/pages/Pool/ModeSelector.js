import React, { useState, useCallback } from 'react'
import { withRouter, NavLink } from 'react-router-dom'
import { useTranslation } from 'react-i18next'
import styled from 'styled-components'

import OversizedPanel from '../../components/OversizedPanel'
import { ReactComponent as Dropdown } from '../../assets/images/dropdown-blue.svg'

import Modal from '../../components/Modal'
import { useBodyKeyDown } from '../../hooks'

import { lighten } from 'polished'

const poolTabOrder = [
  {
    path: '/add-liquidity',
    textKey: 'addLiquidity',
    regex: /\/add-liquidity/
  },
  {
    path: '/remove-liquidity',
    textKey: 'removeLiquidity',
    regex: /\/remove-liquidity/
  },
  {
    path: '/create-exchange',
    textKey: 'createExchange',
    regex: /\/create-exchange.*/
  }
]

const LiquidityContainer = styled.div`
  ${({ theme }) => theme.flexRowNoWrap};
  align-items: center;
  padding: 1rem 1rem;
  font-size: 1rem;
  color: ${({ theme }) => theme.royalBlue};
  font-weight: 500;
  cursor: pointer;

  :hover {
    color: ${({ theme }) => lighten(0.1, theme.royalBlue)};
  }

  img {
    height: 0.75rem;
    width: 0.75rem;
  }
`

const LiquidityLabel = styled.span`
  flex: 1 0 auto;
`

const activeClassName = 'MODE'

const StyledNavLink = styled(NavLink).attrs({
  activeClassName
})`
  ${({ theme }) => theme.flexRowNoWrap}
  padding: 1rem;
  margin-left: 1rem;
  margin-right: 1rem;
  font-size: 1rem;
  cursor: pointer;
  text-decoration: none;
  color: ${({ theme }) => theme.doveGray};
  font-size: 1rem;

  &.${activeClassName} {
    background-color: ${({ theme }) => theme.inputBackground};
    border-radius: 3rem;
    border: 1px solid ${({ theme }) => theme.mercuryGray};
    font-weight: 500;
    color: ${({ theme }) => theme.royalBlue};
  }
`

const PoolModal = styled.div`
  background-color: ${({ theme }) => theme.inputBackground};
  width: 100%;
  height: 100%;
  padding: 2rem 0 2rem 0;
`

const WrappedDropdown = ({ isError, highSlippageWarning, ...rest }) => <Dropdown {...rest} />
const ColoredDropdown = styled(WrappedDropdown)`
  path {
    stroke: ${({ theme }) => theme.royalBlue};
  }
`

function ModeSelector({ location: { pathname }, history }) {
  const { t } = useTranslation()

  const [modalIsOpen, setModalIsOpen] = useState(false)

  const activeTabKey = poolTabOrder[poolTabOrder.findIndex(({ regex }) => pathname.match(regex))].textKey

  const navigate = useCallback(
    direction => {
      const tabIndex = poolTabOrder.findIndex(({ regex }) => pathname.match(regex))
      history.push(poolTabOrder[(tabIndex + poolTabOrder.length + direction) % poolTabOrder.length].path)
    },
    [pathname, history]
  )
  const navigateRight = useCallback(() => {
    navigate(1)
  }, [navigate])
  const navigateLeft = useCallback(() => {
    navigate(-1)
  }, [navigate])

  useBodyKeyDown('ArrowDown', navigateRight, modalIsOpen)
  useBodyKeyDown('ArrowUp', navigateLeft, modalIsOpen)

  return (
    <OversizedPanel hideTop>
      <LiquidityContainer
        onClick={() => {
          setModalIsOpen(true)
        }}
      >
        <LiquidityLabel>{t(activeTabKey)}</LiquidityLabel>
        <ColoredDropdown alt="arrow down" />
      </LiquidityContainer>
      <Modal
        isOpen={modalIsOpen}
        onDismiss={() => {
          setModalIsOpen(false)
        }}
      >
        <PoolModal>
          {poolTabOrder.map(({ path, textKey, regex }) => (
            <StyledNavLink
              key={path}
              to={path}
              isActive={(_, { pathname }) => pathname.match(regex)}
              onClick={() => {
                setModalIsOpen(false)
              }}
            >
              {t(textKey)}
            </StyledNavLink>
          ))}
        </PoolModal>
      </Modal>
    </OversizedPanel>
  )
}

export default withRouter(ModeSelector)
