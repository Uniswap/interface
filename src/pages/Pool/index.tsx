import React, { useContext, useMemo } from 'react'
import styled, { ThemeContext } from 'styled-components'
import { Pair } from '@uniswap/sdk'
import { Link } from 'react-router-dom'
import { SwapPoolTabs } from '../../components/NavigationTabs'

import FullPositionCard from '../../components/PositionCard'
import { useUserHasLiquidityInAllTokens } from '../../data/V1'
import { useTokenBalancesWithLoadingIndicator } from '../../state/wallet/hooks'
import { StyledInternalLink, ExternalLink, TYPE, HideSmall } from '../../theme'
import { Text } from 'rebass'
import Card from '../../components/Card'
import { RowBetween, RowFixed } from '../../components/Row'
import { ButtonPrimary, ButtonSecondary } from '../../components/Button'
import { AutoColumn } from '../../components/Column'

import { useActiveWeb3React } from '../../hooks'
import { usePairs } from '../../data/Reserves'
import { toV2LiquidityToken, useTrackedTokenPairs } from '../../state/user/hooks'
import { Dots } from '../../components/swap/styleds'
import { CardSection, DataCard, CardNoise, CardBGImage } from '../../components/earn/styled'

const PageWrapper = styled(AutoColumn)`
  max-width: 640px;
  width: 100%;
`

const VoteCard = styled(DataCard)`
  background: radial-gradient(76.02% 75.41% at 1.84% 0%, #27ae60 0%, #000000 100%);
  overflow: hidden;
`

const TitleRow = styled(RowBetween)`
  ${({ theme }) => theme.mediaWidth.upToSmall`
    flex-wrap: wrap;
    gap: 12px;
    width: 100%;
    flex-direction: column-reverse;
  `};
`

const ButtonRow = styled(RowFixed)`
  gap: 8px;
  ${({ theme }) => theme.mediaWidth.upToSmall`
    width: 100%;
    flex-direction: row-reverse;
    justify-content: space-between;
  `};
`

const ResponsiveButtonPrimary = styled(ButtonPrimary)`
  width: fit-content;
  ${({ theme }) => theme.mediaWidth.upToSmall`
    width: 48%;
  `};
`

const ResponsiveButtonSecondary = styled(ButtonSecondary)`
  width: fit-content;
  ${({ theme }) => theme.mediaWidth.upToSmall`
    width: 48%;
  `};
`

const EmptyProposals = styled.div`
  border: 1px solid ${({ theme }) => theme.text4};
  padding: 16px 12px;
  border-radius: 12px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`

export default function Pool() {
  const theme = useContext(ThemeContext)
  const { account } = useActiveWeb3React()

  // fetch the user's balances of all tracked V2 LP tokens
  const trackedTokenPairs = useTrackedTokenPairs()
  const tokenPairsWithLiquidityTokens = useMemo(
    () => trackedTokenPairs.map(tokens => ({ liquidityToken: toV2LiquidityToken(tokens), tokens })),
    [trackedTokenPairs]
  )
  const liquidityTokens = useMemo(() => tokenPairsWithLiquidityTokens.map(tpwlt => tpwlt.liquidityToken), [
    tokenPairsWithLiquidityTokens
  ])
  const [v2PairsBalances, fetchingV2PairBalances] = useTokenBalancesWithLoadingIndicator(
    account ?? undefined,
    liquidityTokens
  )

  // fetch the reserves for all V2 pools in which the user has a balance
  const liquidityTokensWithBalances = useMemo(
    () =>
      tokenPairsWithLiquidityTokens.filter(({ liquidityToken }) =>
        v2PairsBalances[liquidityToken.address]?.greaterThan('0')
      ),
    [tokenPairsWithLiquidityTokens, v2PairsBalances]
  )

  const v2Pairs = usePairs(liquidityTokensWithBalances.map(({ tokens }) => tokens))
  const v2IsLoading =
    fetchingV2PairBalances || v2Pairs?.length < liquidityTokensWithBalances.length || v2Pairs?.some(V2Pair => !V2Pair)

  const allV2PairsWithLiquidity = v2Pairs.map(([, pair]) => pair).filter((v2Pair): v2Pair is Pair => Boolean(v2Pair))

  const hasV1Liquidity = useUserHasLiquidityInAllTokens()

  return (
    <>
      <PageWrapper>
        <SwapPoolTabs active={'pool'} />

        <VoteCard>
          <CardBGImage />
          <CardNoise />
          <CardSection>
            <AutoColumn gap="md">
              <RowBetween>
                <TYPE.white fontWeight={600}>Liquidity provider rewards</TYPE.white>
              </RowBetween>
              <RowBetween>
                <TYPE.white fontSize={14}>
                  {`Liquidity providers earn a 0.3% fee on all trades proportional to their share of the pool. Fees are added to the pool, accrue in real time and can be claimed by withdrawing your liquidity.`}
                </TYPE.white>
              </RowBetween>
              <ExternalLink
                style={{ color: 'white', textDecoration: 'underline' }}
                target="_blank"
                href="https://uniswap.org/docs/v2/core-concepts/pools/"
              >
                <TYPE.white fontSize={14}>Read more about providing liquidity</TYPE.white>
              </ExternalLink>
            </AutoColumn>
          </CardSection>
          <CardBGImage />
          <CardNoise />
        </VoteCard>

        <AutoColumn gap="lg" justify="center">
          <AutoColumn gap="lg" style={{ width: '100%' }}>
            <TitleRow style={{ marginTop: '1rem' }} padding={'0'}>
              <HideSmall>
                <TYPE.mediumHeader style={{ marginTop: '0.5rem', justifySelf: 'flex-start' }}>
                  Your liquidity
                </TYPE.mediumHeader>
              </HideSmall>
              <ButtonRow>
                <ResponsiveButtonSecondary as={Link} padding="6px 8px" to="/create/ETH">
                  Create a pair
                </ResponsiveButtonSecondary>
                <ResponsiveButtonPrimary id="join-pool-button" as={Link} padding="6px 8px" to="/add/ETH">
                  <Text fontWeight={500} fontSize={16}>
                    Add Liquidity
                  </Text>
                </ResponsiveButtonPrimary>
              </ButtonRow>
            </TitleRow>

            {!account ? (
              <Card padding="40px">
                <TYPE.body color={theme.text3} textAlign="center">
                  Connect to a wallet to view your liquidity.
                </TYPE.body>
              </Card>
            ) : v2IsLoading ? (
              <EmptyProposals>
                <TYPE.body color={theme.text3} textAlign="center">
                  <Dots>Loading</Dots>
                </TYPE.body>
              </EmptyProposals>
            ) : allV2PairsWithLiquidity?.length > 0 ? (
              <>
                <ButtonSecondary>
                  <RowBetween>
                    <ExternalLink href={'https://uniswap.info/account/' + account}>
                      Account analytics and accrued fees
                    </ExternalLink>
                    <span> â†—</span>
                  </RowBetween>
                </ButtonSecondary>

                {allV2PairsWithLiquidity.map(v2Pair => (
                  <FullPositionCard key={v2Pair.liquidityToken.address} pair={v2Pair} />
                ))}
              </>
            ) : (
              <EmptyProposals>
                <TYPE.body color={theme.text3} textAlign="center">
                  No liquidity found.
                </TYPE.body>
              </EmptyProposals>
            )}

            <AutoColumn justify={'center'} gap="md">
              <Text textAlign="center" fontSize={14} style={{ padding: '.5rem 0 .5rem 0' }}>
                {hasV1Liquidity ? 'Uniswap V1 liquidity found!' : "Don't see a pool you joined?"}{' '}
                <StyledInternalLink id="import-pool-link" to={hasV1Liquidity ? '/migrate/v1' : '/find'}>
                  {hasV1Liquidity ? 'Migrate now.' : 'Import it.'}
                </StyledInternalLink>
              </Text>
            </AutoColumn>
          </AutoColumn>
        </AutoColumn>
      </PageWrapper>
    </>
  )
}
