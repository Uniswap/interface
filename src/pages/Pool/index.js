import React, { Suspense, lazy, useEffect } from 'react'
import ReactGA from 'react-ga'
import { Switch, Route, Redirect } from 'react-router-dom'
import ModeSelector from './ModeSelector'

const AddLiquidity = lazy(() => import('./AddLiquidity'))
const RemoveLiquidity = lazy(() => import('./RemoveLiquidity'))
const CreateExchange = lazy(() => import('./CreateExchange'))

export default function Pool({ params }) {
  useEffect(() => {
    ReactGA.pageview(window.location.pathname + window.location.search)
  }, [])

  const AddLiquidityParams = () => <AddLiquidity params={params} />

  const RemoveLiquidityParams = () => <RemoveLiquidity params={params} />

  const CreateExchangeParams = () => <CreateExchange params={params} />

  return (
    <>
      <ModeSelector />
      {/* this Suspense is for route code-splitting */}
      <Suspense fallback={null}>
        <Switch>
          <Route exact strict path="/add-liquidity" component={AddLiquidityParams} />
          <Route exact strict path="/remove-liquidity" component={RemoveLiquidityParams} />
          <Route exact strict path="/create-exchange" component={CreateExchangeParams} />
          <Route
            path="/create-exchange/:tokenAddress"
            render={({ match }) => {
              return (
                <Redirect to={{ pathname: '/create-exchange', state: { tokenAddress: match.params.tokenAddress } }} />
              )
            }}
          />
          <Redirect to="/add-liquidity" />
        </Switch>
      </Suspense>
    </>
  )
}
