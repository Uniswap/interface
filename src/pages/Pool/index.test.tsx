import { render, act } from 'test-utils'
import Pool from '.'
import * as switchChain from 'utils/switchChain'
import * as useV3Positions from 'hooks/useV3Positions'
import { BrowserRouter as Router } from 'react-router-dom'

jest.mock('@web3-react/core', () => {
  const web3React = jest.requireActual('@web3-react/core')
  return {
    ...web3React,
    useWeb3React: () => ({
      chainId: 1,
    }),
  }
})

describe('networks', () => {
  it('renders error card when unsupported chain is selected', () => {
    jest.spyOn(switchChain, 'isChainAllowed').mockReturnValue(false)
    jest.spyOn(useV3Positions, 'useV3Positions').mockImplementation(() => {
      return { loading: false, positions: undefined }
    })

    act(() => {
      const { asFragment } = render(
        <Router>
          <Pool />
        </Router>
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })

  it('renders empty positions card when on supported chain with no positions', () => {
    jest.spyOn(switchChain, 'isChainAllowed').mockReturnValue(true)
    jest.spyOn(useV3Positions, 'useV3Positions').mockImplementation(() => {
      return { loading: false, positions: undefined }
    })

    act(() => {
      const { asFragment } = render(
        <Router>
          <Pool />
        </Router>
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
