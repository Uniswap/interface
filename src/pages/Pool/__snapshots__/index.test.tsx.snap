// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`networks renders empty positions card when on supported chain with no positions 1`] = `
<DocumentFragment>
  <div
    class="Column__AutoColumn-sc-1r2yyln-2 Pool__PageWrapper-sc-um62p3-0 fYeWeg eKrzNy"
  >
    <div
      class="NavigationTabs__Tabs-sc-17q0q28-0 kdXLVp"
      style="margin-bottom: 20px; display: none; padding: 1rem 1rem 0px 1rem;"
    >
      <a
        class="NavigationTabs__StyledNavLink-sc-17q0q28-1 fmMEMN"
        href="/swap"
        id="swap-nav-link"
      >
        Swap
      </a>
      <a
        aria-current="page"
        class="NavigationTabs__StyledNavLink-sc-17q0q28-1 fmMEMN ACTIVE"
        href="/pool"
        id="pool-nav-link"
      >
        Pool
      </a>
    </div>
    <div
      class="Column__AutoColumn-sc-1r2yyln-2 bPeoqt"
    >
      <div
        class="Column__AutoColumn-sc-1r2yyln-2 ksWoxm"
        style="width: 100%;"
      >
        <div
          class="sc-bdnxRM Row-sc-u7azg8-0 Row__RowBetween-sc-u7azg8-1 Pool__TitleRow-sc-um62p3-1 dHNyFh xDbmg hzJkYd hCbTuX"
          style="margin-top: 1rem;"
        >
          <div
            class="theme__TextWrapper-sc-5lu8um-0 etkwRL css-1xd3sow"
          >
            Pools Overview
          </div>
          <div
            class="sc-bdnxRM Row-sc-u7azg8-0 Row__RowFixed-sc-u7azg8-4 Pool__ButtonRow-sc-um62p3-2 lmTMKd hLLNig kraLww cObwhm"
          >
            <div
              class="Menu__StyledMenu-sc-1s7r6xc-3 kAixpC Pool__Menu-sc-um62p3-3 hWucEK"
            >
              <button
                class="sc-bdnxRM hSDgzJ Button__BaseButton-sc-tdn257-0 Button__ButtonGray-sc-tdn257-3 Pool__MoreOptionsButton-sc-um62p3-5 bVCaPv bFJueL eTKJIV"
              >
                <div
                  class="theme__TextWrapper-sc-5lu8um-0 etkwRL Pool__MoreOptionsText-sc-um62p3-6 dqQZwl css-1frkxb8"
                >
                  More
                  <svg
                    fill="none"
                    height="15"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="15"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <polyline
                      points="6 9 12 15 18 9"
                    />
                  </svg>
                </div>
              </button>
            </div>
            <a
              class="Button__BaseButton-sc-tdn257-0 Button__ButtonPrimary-sc-tdn257-1 Pool__ResponsiveButtonPrimary-sc-um62p3-10 bVCaPv jpiSkz gVNiJq"
              href="/add/ETH"
              id="join-pool-button"
            >
              + New Position
            </a>
          </div>
        </div>
        <main
          class="Pool__MainContentWrapper-sc-um62p3-11 fUxNHF"
        >
          <div
            class="Pool__ErrorContainer-sc-um62p3-7 frTKSy"
          >
            <div
              class="theme__TextWrapper-sc-5lu8um-0 chxxqs css-9fgfz0"
            >
              <svg
                class="Pool__InboxIcon-sc-um62p3-9 fDcSZm"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="22 12 16 12 14 15 10 15 8 12 2 12"
                />
                <path
                  d="M5.45 5.11L2 12v6a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2v-6l-3.45-6.89A2 2 0 0 0 16.76 4H7.24a2 2 0 0 0-1.79 1.11z"
                />
              </svg>
              <div>
                Your active V3 liquidity positions will appear here.
              </div>
            </div>
            <button
              class="sc-bdnxRM hSDgzJ Button__BaseButton-sc-tdn257-0 Button__ButtonPrimary-sc-tdn257-1 bVCaPv jpiSkz"
              style="margin-top: 2em; padding: 8px 16px;"
            >
              Connect a wallet
            </button>
          </div>
        </main>
        <span
          class="components__HideSmall-sc-88web0-13 dZaAKV"
        >
          <section
            class="CTACards__CTASection-sc-1lbpl6-0 cIXeJs"
          >
            <a
              class="components__StyledLink-sc-88web0-5 kYFdcM CTACards__CTA1-sc-1lbpl6-1 eofYCX"
              href="https://help.uniswap.org/en/articles/5391541-providing-liquidity-on-uniswap-v3"
              rel="noopener noreferrer"
              target="_blank"
            >
              <div
                class="Column__AutoColumn-sc-1r2yyln-2 CTACards__ResponsiveColumn-sc-1lbpl6-4 fYeWeg gjzWtJ"
              >
                <div
                  class="theme__TextWrapper-sc-5lu8um-0 etkwRL CTACards__HeaderText-sc-1lbpl6-3 kWiYgr css-kh5b9v"
                >
                  Learn about providing liquidity ↗
                </div>
                <div
                  class="theme__TextWrapper-sc-5lu8um-0 etkwRL css-1frkxb8"
                  style="align-items: center; display: flex;"
                >
                  Check out our v3 LP walkthrough and migration guides.
                </div>
              </div>
            </a>
            <a
              class="components__StyledLink-sc-88web0-5 kYFdcM CTACards__CTA2-sc-1lbpl6-2 fbVxQY"
              href="https://info.uniswap.org/#/pools"
              rel="noopener noreferrer"
              target="_blank"
            >
              <div
                class="Column__AutoColumn-sc-1r2yyln-2 CTACards__ResponsiveColumn-sc-1lbpl6-4 fYeWeg gjzWtJ"
              >
                <div
                  class="theme__TextWrapper-sc-5lu8um-0 etkwRL CTACards__HeaderText-sc-1lbpl6-3 kWiYgr css-kh5b9v"
                  style="align-self: flex-start;"
                >
                  Top pools ↗
                </div>
                <div
                  class="theme__TextWrapper-sc-5lu8um-0 etkwRL css-1frkxb8"
                  style="align-self: flex-start;"
                >
                  Explore Uniswap Analytics.
                </div>
              </div>
            </a>
          </section>
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`networks renders error card when unsupported chain is selected 1`] = `
<DocumentFragment>
  <div
    class="Column__AutoColumn-sc-1r2yyln-2 Pool__PageWrapper-sc-um62p3-0 fYeWeg eKrzNy"
  >
    <div
      class="NavigationTabs__Tabs-sc-17q0q28-0 kdXLVp"
      style="margin-bottom: 20px; display: none; padding: 1rem 1rem 0px 1rem;"
    >
      <a
        class="NavigationTabs__StyledNavLink-sc-17q0q28-1 fmMEMN"
        href="/swap"
        id="swap-nav-link"
      >
        Swap
      </a>
      <a
        aria-current="page"
        class="NavigationTabs__StyledNavLink-sc-17q0q28-1 fmMEMN ACTIVE"
        href="/pool"
        id="pool-nav-link"
      >
        Pool
      </a>
    </div>
    <div
      class="Column__AutoColumn-sc-1r2yyln-2 bPeoqt"
    >
      <div
        class="Column__AutoColumn-sc-1r2yyln-2 ksWoxm"
        style="width: 100%;"
      >
        <div
          class="sc-bdnxRM Row-sc-u7azg8-0 Row__RowBetween-sc-u7azg8-1 Pool__TitleRow-sc-um62p3-1 dHNyFh xDbmg hzJkYd hCbTuX"
          style="margin-top: 1rem;"
        >
          <div
            class="theme__TextWrapper-sc-5lu8um-0 etkwRL css-1xd3sow"
          >
            Pools Overview
          </div>
        </div>
        <main
          class="Pool__MainContentWrapper-sc-um62p3-11 fUxNHF"
        >
          <div
            class="Pool__ErrorContainer-sc-um62p3-7 frTKSy"
          >
            <div
              class="theme__TextWrapper-sc-5lu8um-0 chxxqs css-9fgfz0"
            >
              <svg
                class="Pool__NetworkIcon-sc-um62p3-8 kvLezX"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="22 12 18 12 15 21 9 3 6 12 2 12"
                />
              </svg>
              <div>
                Your connected network is unsupported. Request support 
                <a
                  class="components__StyledLink-sc-88web0-5 kYFdcM"
                  href="https://uniswap.canny.io/feature-requests"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  here
                </a>
                .
              </div>
            </div>
          </div>
        </main>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
